File: cluster/extensions-scaladsl/src/main/scala/com/lightbend/lagom/scaladsl/cluster/projections/ProjectorRegistry.scala
/*
 * Copyright (C) 2016-2019 Lightbend Inc. <https://www.lightbend.com>
 */

package com.lightbend.lagom.scaladsl.cluster.projections

import com.lightbend.lagom.internal.cluster.projections.ProjectorRegistryImpl
import com.lightbend.lagom.internal.cluster.projections.ProjectorRegistryImpl.ProjectionMetadata
import com.lightbend.lagom.internal.cluster.projections.ProjectorRegistryImpl.ProjectorStatus

import scala.concurrent.Future

/** TODO: docs
 *
 */
class ProjectorRegistry(impl:ProjectorRegistryImpl) {

  def getStatus: Future[Map[ProjectionMetadata, ProjectorStatus]] =
    impl.getStatus()

  // TODO: impl stop
  // TODO: impl start

}

File: cluster/extensions-javadsl/src/main/java/com/lightbend/lagom/javadsl/cluster/projections/ProjectorRegistry.java
/*
 * Copyright (C) 2016-2019 Lightbend Inc. <https://www.lightbend.com>
 */

package com.lightbend.lagom.javadsl.cluster.projections;

import com.lightbend.lagom.internal.cluster.projections.ProjectorRegistryImpl;

import java.util.Map;
import java.util.concurrent.CompletionStage;

/** TODO: docs */
public interface ProjectorRegistry {

  CompletionStage<
          Map<ProjectorRegistryImpl.ProjectionMetadata, ProjectorRegistryImpl.ProjectorStatus>>
      getStatus();

  // TODO: implement stop
  // TODO: implement start

}
